platform/linux-generic/odp_queue.c

```diff
@@ -471,51 +476,18 @@ static inline int deq_multi(queue_entry_t *queue, odp_buffer_hdr_t *buf_hdr[],
 		}
 
 		UNLOCK(&queue->s.lock);
-		return 0;
-	}
-
-	for (i = 0; i < num && hdr; ) {
-		int burst_num = hdr->burst_num;
-		int first     = hdr->burst_first;
 
-		/* First, get bursted buffers */
-		for (j = 0; j < burst_num && i < num; j++, i++) {
-			buf_hdr[i] = hdr->burst[first + j];
-			odp_prefetch(buf_hdr[i]);
-		}
-
-		if (burst_num) {
-			hdr->burst_num   = burst_num - j;
-			hdr->burst_first = first + j;
-		}
-
-		if (i == num)
-			break;
-
-		/* When burst is empty, consume the current buffer header and
-		 * move to the next header */
-		buf_hdr[i] = hdr;
-		next       = hdr->next;
-		hdr->next  = NULL;
-		hdr        = next;
-		updated++;
-		i++;
+		return 0;
 	}
 
-	/* Write head only if updated */
-	if (updated)
-		queue->s.head = hdr;
-
-	/* Queue is empty */
-	if (hdr == NULL)
-		queue->s.tail = NULL;
-
 	if (status_sync && queue->s.type == ODP_QUEUE_TYPE_SCHED)
 		sched_fn->save_context(queue->s.index);
 
 	UNLOCK(&queue->s.lock);
 
-	return i;
+	buffer_index_to_buf(buf_hdr, buf_idx, num_deq);
```

Comment:
Given its name, this looks like an extraneous statement that should be deleted. Renaming this to something like `prefetch_dequeued_bufs()` would make the intent clearer here.

Reply:
This function converts 32 bit buffer indexes to buffer header pointers. The counter operation is buffer_index_from_buf(). The prefetch is a side effect of the function, which may be changed/moved any time if it's found out that there's a place for prefetching. I actually plan to test if number of prefetches should be limited as e.g. 32 consecutive prefetches may be too much for some CPU architectures.

Reply:
That functionality is not obvious from the name. It either implies that one of the input arguments is written (not true here) or the reader might assume that it is an expression without side-effect and should be deleted (what I originally thought when reading it). You should pick a routine name that makes it clear it's actually doing something real, in this case performing prefetch processing.